### ISSUE TITLE
Bad attribute access when assertion contains empty AttributeValues

### ISSUE THREAD

**Issue by dpatti** [Clever/saml2#38]
I noticed that OneLogin sends login assertions like this if there is no value for an attribute:

```xml
<saml:Attribute Name="NoValueAttr" NameFormat="urn:oasis:names:tc:SAML:2.0:attrname-format:basic">
  <saml:AttributeValue xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:type="xs:string"/>
</saml:Attribute>
```

However, this causes an error when parsing attribute values. I am getting a `TypeError: Cannot read property 'data' of undefined` on [this line](https://github.com/Clever/saml2/blob/master/lib/saml2.coffee#L287). The `attribute_value` in question has a `childNodes` object of length 0. This can be fixed with a `filter()` to get rid of these empty attributes values, but since my knowledge of SAML is lacking, I wanted to make sure this was the right way to approach this problem.

**Comment by azylman** [comment]
Thanks for the report! They're supposed to count as empty values, so instead of filtering them we're treating them as an empty string in #39. Should be published today or tomorrow.

**Comment by dpatti** [comment]
Oh fantastic. Thanks, I really appreciate the fast turnaround!

**Comment by azylman** [comment]
Should be published as 1.0.2

**Comment by dpatti** [comment]
The new version works great. Thanks again for the fix and the library. It's been a huge help.
